src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -415,13 +416,13 @@ private void Compact(long removalSizeTarget, Func<CacheEntry, long> computeEntry
                     switch (entry.Priority)
                     {
                         case CacheItemPriority.Low:
-                            lowPriEntries.Add(entry);
+                            lowPriEntries.Add(new CompactPriorityEntry(entry, entry.LastAccessed));
```

Comment:
I assume there will not be any boxing operation that causes additional allocations. right?

Reply:
Correct. We are using a `List<CompactPriorityEntry>`, so no boxing will occur. The small structs will just be copied in and out of the list. This is the same with the 7.0+ implementation, but there we are using ValueTuple instead of a custom struct.